### PR TITLE
[LOOP-3838] Fix OnboardingManager crash

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-3838
- Ensure LoopAppManager launch is not invoked multiple times

### CODE REVIEW

File: Loop/Managers/LoopAppManager.swift [hunk 1]
```diff
@@ -119,10 +115,26 @@ class LoopAppManager: NSObject {
         }
     }
 
+    private func checkProtectedDataAvailable() {
+        dispatchPrecondition(condition: .onQueue(.main))
+        precondition(state == .checkProtectedDataAvailable)
+
+        guard isProtectedDataAvailable() else {
+            log.default("Protected data not available; deferring launch...")
+            return
```
Comment (rickpasetto) [inline Loop/Managers/LoopAppManager.swift:124]: Just trying to understand the state machine here.  If this happens, what calls `launch()` (or `resumeLaunch()`) in this case?

**Review by rickpasetto — COMMENTED** [review]
Looks good, but I have a question about the state machine...

File: Loop/Managers/LoopAppManager.swift (hunk 1 above)
Comment (darinkrauss) [reply Loop/Managers/LoopAppManager.swift:124]: `AppDelegate.applicationProtectedDataDidBecomeAvailable` does.

**Review by rickpasetto — APPROVED** [review]
Makes sense. LGTM.

**Pull request merged**